fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MvRewritePreprocessor.java

```diff
@@ -153,6 +172,81 @@ private void logMVParams(ConnectContext connectContext, Set<Table> queryTables)
                 sessionVariable.getMaterializedViewRewriteMode());
         logMVPrepare(connectContext, "---------------------------------");
     }
+
+    private void processPlanWithView(ConnectContext connectContext,
+                                     OptExpression logicOperatorTree,
+                                     ColumnRefFactory columnRefFactory,
+                                     ColumnRefSet requiredColumns) {
+        Map<LogicalViewScanOperator, OptExpression> viewPlanMap = Maps.newHashMap();
+        // process equivalent operator，construct logical plan with view
+        OptExpression logicalPlanWithView = extractLogicalPlanWithView(logicOperatorTree, viewPlanMap);
+        if (viewPlanMap.isEmpty()) {
+            // means there is no plan with view
+            return;
+        }
+        // optimize logical plan with view
+        OptExpression optimizedPlan = optimizeViewPlan(
+                logicalPlanWithView, connectContext, requiredColumns, columnRefFactory);
+        context.setLogicalTreeWithView(optimizedPlan);
+
+        Map<LogicalViewScanOperator, OptExpression> optimizedViewPlanMap = Maps.newHashMap();
+        for (LogicalViewScanOperator viewScanOperator : viewPlanMap.keySet()) {
+            OptExpression viewLogicalTree = viewPlanMap.get(viewScanOperator);
+            // optimize logical tree of view and keep them in OptimizerContext,
+            // which will be used in union rewrite or only some views are rewritten
+            OptExpression optimizedViewPlan = optimizeViewPlan(
+                    viewLogicalTree, connectContext, viewScanOperator.getOutputColumnSet(), columnRefFactory);
+            optimizedViewPlanMap.put(viewScanOperator, optimizedViewPlan);
+        }
+        context.setViewPlanMap(optimizedViewPlanMap);
```

Comment:
why we need invoke optimizeViewPlan twice and Context needs keep a optimizedViewPlan and viewPlanMap